org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/breakpoints/ScalaDebugBreakpointManager.scala

```diff
@@ -88,6 +96,8 @@ private[debug] object ScalaDebugBreakpointManagerActor {
   case class BreakpointAdded(breakpoint: IBreakpoint)
   case class BreakpointRemoved(breakpoint: IBreakpoint)
   case class BreakpointChanged(breakpoint: IBreakpoint, delta: IMarkerDelta)
+  case class ReenableBreakpointsAfterHcr(classNames: Seq[String])
+  case object ReenableBreakpointAfterHcr
```

Comment:
Both, please :) Since actors are untyped, we need to use conventions, and I think it makes is much easier to group messages in the companion object of the actor that handles them. A one-line description is also useful, but not always necessary. I leave it up to do you decide if it's needed.